syz-manager/manager.go

```diff
@@ -261,7 +262,26 @@ func RunManager(cfg *mgrconfig.Config) {
 	log.Logf(0, "serving rpc on tcp://%v", mgr.serv.Port)
 
 	if cfg.DashboardAddr != "" {
-		dash, err := dashapi.New(cfg.DashboardClient, cfg.DashboardAddr, cfg.DashboardKey)
```

Comment:
Some dash.SetUserAgent() call after this line may look better.

Reply:
That's what I wanted to do initially. However, I saw that the `RequestCtor` is used inside `NewCustom` (and by extension `New`). So A request could be made inside the constructors, which means that the correct user-agent should already be set.

However from what I see this only happens if `key == ""` which in turn implies that this is done in GCE, which might mean that a proxy is probably not used. So using the `SetUserAgent` method later is fine.

We could also move the call to `MakeCache` in a separate function, to be called after `New`. But that will make the API exposed by `dashapi.go` a bit more complex.

Reply:
Will some hardcoded value work for you?
Something like "syz-manager".

Reply:
We can use optional arguments in `dashapi.New` for this purpose, like e.g. we do here:

https://github.com/google/syzkaller/blob/7d7fe8807550c6e02def380d335ae0230f1e495c/vm/vm.go#L193-L207

https://github.com/google/syzkaller/blob/7d7fe8807550c6e02def380d335ae0230f1e495c/syz-manager/manager.go#L923-L930

Reply:
Updateed the code. Let me know if it looks good.